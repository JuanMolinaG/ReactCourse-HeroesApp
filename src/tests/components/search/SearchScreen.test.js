import React from 'react';
import { mount } from 'enzyme';
import { SearchScreen } from '../../../components/search/SearchScreen';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Pruebas en <SearchScreen />', () => {
  test('Debe de mostrar el componente correctamente', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <Route path='/search' component={SearchScreen}></Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero');
  });

  test('Debe de mostrar a Batman y el input debe tener el valor del query', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <Route path='/search' component={SearchScreen}></Route>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input').prop('value')).toBe('batman');
  });

  test('Debe de mostrar un mensaje de error si no se encuentra el hero', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/search?q=batsy']}>
        <Route path='/search' component={SearchScreen}></Route>
      </MemoryRouter>
    );

    expect(wrapper.find('.alert-danger').text().trim()).toBe(
      'There is no a hero called batsy'
    );
  });

  test('Debe de llamar el push del history', () => {
    const historyMock = {
      push: jest.fn(),
    };

    const wrapper = mount(
      <MemoryRouter initialEntries={['/search']}>
        <Route
          path='/search'
          component={() => <SearchScreen history={historyMock} />}
        ></Route>
      </MemoryRouter>
    );

    wrapper
      .find('input')
      .simulate('change', { target: { name: 'search', value: 'Batman' } });

    wrapper.find('form').prop('onSubmit')({ preventDefault() {} });

    expect(historyMock.push).toHaveBeenCalledWith('?q=Batman');
  });
});
