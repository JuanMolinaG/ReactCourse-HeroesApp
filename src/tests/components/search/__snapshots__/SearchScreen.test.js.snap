// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> Debe de mostrar a Batman y el input debe tener el valor del query 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <form
        autoComplete="off"
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="search"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn mt-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="row"
      >
        <div
          className="card-container col-md-4 col-sm-6 mb-4"
        >
          <div
            className="card-flip"
          >
            <div
              className="card front"
            >
              <img
                alt="Batman"
                className="card-img"
                src="./assets/heroes/dc-batman.jpg"
              />
            </div>
            <div
              className="card back p-4"
            >
              <div
                className="card-block"
              >
                <h5
                  className="card-title"
                >
                  Batman
                </h5>
                <p
                  className="card-text"
                >
                  <b>
                    Alter Ego:
                  </b>
                   
                  Bruce Wayne
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    <b>
                      First Appeatance:
                    </b>
                     
                    Detective Comics #27
                  </small>
                </p>
                <a
                  href="/hero/dc-batman"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchScreen /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <form
        autoComplete="off"
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="search"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn mt-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="row"
      >
        <div
          className="alert alert-info col-12"
        >
          Search a hero
        </div>
      </div>
    </div>
  </div>
</div>
`;
